{#include index}
<main class="form-signin w-100 m-auto">
    <div id="failureAlert" class="alert alert-danger d-none" role="alert">Authentication failed</div>
    <form id="loginForm">
        <h1 class="h3 mb-3 fw-normal">Please sign in</h1>
        <div class="form-floating"> 
            <input name="j_username" id="j_username" type="text" class="form-control" placeholder="name@example.com"> 
            <label for="floatingInput">Email address</label> 
        </div>
        <div class="form-floating"> 
            <input name="j_password" id="j_password" type="password" class="form-control"  placeholder="Password">
            <label for="floatingPassword">Password</label>
        </div>
        <button class="btn btn-primary w-100 py-2" type="submit">Sign in</button>
    </form>
</main>
<script>
    document.getElementById("loginForm").onsubmit = async (event) => {
        // stop the form submitting itself
        event.preventDefault();

        // Make an HTTP POST request using fetch against j_security_check endpoint
        fetch('/j_security_check', {
            method: 'POST',
            body: new FormData(event.target)
        }).then((response) => {
                if (response.ok) {
                    console.error("Successfully signed in");

                    // Authentication was successful
                    // Get the url params
                    const urlParams = new URLSearchParams(window.location.search);

                    // Get the redirect_uri if avaiable else default to "/"
                    const redirectUri = urlParams.get('redirect_uri') ? urlParams.get('redirect_uri') : "/";

                    // redirect to the uri
                    window.location.href = redirectUri;
                } else {
                    // Authentication failed
                    console.error("Invalid credentials");
                    
                    // Show the alert
                    document.getElementById("failureAlert").classList.remove('d-none')
                }
            })
            .catch((error) => {
                console.error(error);
            });
    }
</script>